### ISSUE TITLE
Netflix "Browser Not Supported"

### ISSUE THREAD

**Issue by oscartbeaumont** [oscartbeaumont/ElectronPlayer#31]
It looks like Netflix have pushed an update to their frontend over the last week that prevents it from working with this app. I assume this is because of the old browser version it is using due to the latest version of Electron not working with Widevine the DRM module used by Netflix.

**Comment by oscartbeaumont** [comment]
It is now fixed!!!!!!! The fix is an absolute mess because now the project has two separate `package.json`'s but it will work for now. I am going to continue trying to gain some traction upstream to fix this issue properly. The fix has also caused a difference in Electron version. The Linux version works on Electron 5.0.4 while the Mac version is on Electron 4.2.4. I am so happy there was a fix I was expecting to have to discard this project as I have read other people having the same problem and having to give up.

**Comment by ivanov35** [comment]
Can you help, how did you fix this problem exactly ?

**Comment by brunoti** [comment]
I'm having this problem in MacOS, first time trying.

![Screenshot 2020-04-26 16 43 23(2)](https://user-images.githubusercontent.com/6011421/80317943-1b136700-87dd-11ea-9c79-1f60df130c4c.png)

**Comment by oscartbeaumont** [comment]
Oh, that sucks Netflix finally killed support for ElectronPlayer on MacOS. The reason ElectronPlayer is using an old browser version (which Netflix now doesn't support) is because the newer browser versions don't allow me to have Widevine VMP which is the DRM used by Netflix and other streaming services. To implement Widevine VMP on Mac or if I wanted to Windows, I would need a certificate from Google which I have requested but am not expecting to get one given they don't give them to small, open source projects. I still haven't received a reply and its been about half a year. [This article](https://blog.samuelmaddock.com/posts/google-widevine-blocked-my-browser/) talks about another Electron app having the same problem.
Note: This isn't an issue on Linux because I think it lets me use the development certificates to play streaming services for some unknown reason.

The only solutions left that I could find (contact me if you can come up with anything better) is to ditch Electron for something on Mac that works with Widevine VPM. I could not find/get working any solution out their but I had an idea of how to make one and I have commenced development on it [here called NativeKit](https://github.com/oscartbeaumont/NativeKit). If you go [here](https://electronplayer-beta.otbeaumont.me/), you can get the NativeKit beta of ElectronPlayer although it is missing most of the ElectronPlayer features until I find the time to build them, which being honest isn't going to happen until really late this year. If anyone is interested in getting involved in its development, please send me an email. This separate app core will also slow and split development on ElectronPlayer because the Mac and Linux version will be on different codebases but I am doing my best to make NativeKit as close to Electron as possible to mitigate this issue as much as possible.

I am also currently in the process of rewriting the Linux version of the app and optimising it for the newer version of Electron it is using which would have caused me to phase out the Mac version of the app anyway relatively soon, Netflix have just speed up this process. Maintaining backwards compatibility for the Mac version has been hurting the maintainability and "polish" (stability and reliability) of the entire app which I was hoping to fix which this change.

So in summary, its out of my control (Thanks Google (not)) and on Mac you will have to use the featureless NativeKit based version of the app which you can get [here](https://electronplayer-beta.otbeaumont.me/).

For reference, issue #20 talks about this although is somewhat outdated.

**Comment by oscartbeaumont** [comment]
> Can you help, how did you fix this problem exactly ?

Sorry it has taken me so long to reply, I had not seen your message until today.
You can fix it by updating your app if your on Linux or getting the NativeKit based version of the app [here](https://electronplayer-beta.otbeaumont.me/) if your on Mac although this version of the app doesn't have most of the features you previously had. This may mean you will need to move back to using Chrome/Firefox. Sorry I just have limited free time to maintain this project.

**Comment by brunoti** [comment]
Hi @oscartbeaumont ! That really sucks =/. I will try to use the NativeKit version and see if I can contribute in any way.

Thank you for the projects and for explaining the issue to me.
